### DISCUSSION TITLE
How can I use PI control to make the 5MW_OC4Semi_WSt_WavesWN_Test25 reach its rated power under steady wind conditions？

### DISCUSSION THREAD

**Discussion by whyworlds2** [OpenFAST/openfast#1931] [Q&A] [answered]
Dear @jjonkman , It's about parameters about PI control to make the 5MW_OC4Semi_WSt_WavesWN_Test25 reach its rated power under steady wind conditions. I use 15m/s steady wind file and the JONSWAP/Pierson-Moskowitz spectrum (irregular) as wave condition, but it seems difficult to stabilize the wind tubine and reach the steady point of 15m/s wind speed. The current parameters are P=0.009, I=0.004, the result are shown in figure below.
<img width="1152" alt="屏幕截图 2023-12-11 155213" src="https://github.com/OpenFAST/openfast/assets/152302484/f4f4b205-2414-4829-90a7-02503e305d17">

**Comment by jjonkman** [comment]
Dear @whyworlds2,

I'm not sure I really understand your question.  I would not expect an exact steady-state condition to be reached with irregular wave excitation.  But your results indicate that your model is is achieving rated power of 5 MW.  

Best regards.

**Reply by whyworlds2** [reply]
Dear @jjonkman, I'm sorry for not describing my problem clearly. The problem is that when the simulation time reaches 600 seconds，the results show that the power and genspeed of the wind turbine have significant fluctuations around the rated value. Can it be considered that the WT has reached stability in this situation? Is there a way to achieve better results for WT？ 

Best regards.
![Uploading 屏幕截图 2023-12-12 123636.png…]()

**Reply by jjonkman** [reply, marked as answer]
Dear @whyworlds2,

 Again, I would say that I would not expect an exact steady-state condition to be reached with irregular wave excitation.

Best regards,
